src/v/pandaproxy/schema_registry/sharded_store.cc

```diff
@@ -133,13 +133,34 @@ sharded_store::make_valid_schema(canonical_schema schema) {
 }
 
 ss::future<sharded_store::has_schema_result>
-sharded_store::get_schema_version(subject_schema schema) {
+sharded_store::get_schema_version(subject_schema schema, normalize norm) {
     // Validate the schema (may throw)
     co_await validate_schema(schema.schema.share());
 
     // Determine if the definition already exists
-    auto map = [&schema](store& s) {
-        return s.get_schema_id(schema.schema.def());
+    auto map = [this, norm, &schema](
+                 const store& s) -> ss::future<std::optional<schema_id>> {
+        for (const auto& [id, def] : s.get_schemas()) {
```

Comment:
You may get iterator invalidation across the scheduling point caused by the co_await, I suggest a copy of the container.